packages/sign-client/src/controllers/engine.ts

```diff
@@ -354,6 +354,29 @@ export class Engine extends IEngine {
 
   // ---------- Private Helpers --------------------------------------- //
 
+  private cleanupDuplicatePairings: EnginePrivate["cleanupDuplicatePairings"] = async (
+    session: SessionTypes.Struct,
+  ) => {
+    try {
+      const pairing = this.client.core.pairing.pairings.get(session.pairingTopic);
+      const allPairings = this.client.core.pairing.pairings.getAll();
+      const duplicates = allPairings.filter(
+        (p) =>
+          p.peerMetadata?.url &&
+          p.peerMetadata?.url === session.self.metadata.url &&
+          p.topic !== pairing.topic,
+      );
+      if (duplicates.length === 0) return;
+      this.client.logger.info(`Cleaning up ${duplicates.length} duplicate pairing(s)`);
+      await Promise.all(
+        duplicates.map((p) => this.client.core.pairing.disconnect({ topic: p.topic })),
```

Comment:
Don’t think this is a huge concern with real world usage, but this could trigger a lot of `wc_pairingDelete` requests in quick succession where pairings with the same peer have accumulated.

Reply:
I've tested with ~20 active pairings and it was working just fine. After the initial cleanup - there will be 1 pairing to be removed only